app/models/services/AuthTokenServiceImpl.scala

```diff
@@ -0,0 +1,75 @@
+package models.services
+
+import java.sql.Timestamp
+import java.time.Instant
+import java.util.UUID
+import java.security.MessageDigest
+import javax.inject.Inject
+
+import models.AuthToken
+import models.daos.AuthTokenDAO
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.language.postfixOps
+import play.api.libs.concurrent.Execution.Implicits._
+
+/**
+ * Handles actions to auth tokens.
+ *
+ * @param authTokenDAO The auth token DAO implementation.
+ */
+class AuthTokenServiceImpl @Inject() (authTokenDAO: AuthTokenDAO) extends AuthTokenService {
+
+  /**
+   * AuthToken hasher
+   *
+   * @return A cryptographic hasher utilizing SHA-256
+   */
+  def sha256Hasher: MessageDigest = MessageDigest.getInstance("SHA-256")
+
+  /**
+   * Creates a new auth token and saves it in the backing store.
+   *
+   * @param userID The user ID for which the token should be created.
+   * @param expiry The duration a token expires.
+   * @return The saved auth token.
+   */
+  def create(userID: UUID, expiry: FiniteDuration = 5 minutes) = {
+    val tokenID = UUID.randomUUID()
+    val hashedTokenID = sha256Hasher.digest(tokenID.toString.getBytes)
+    val token = AuthToken(hashedTokenID, userID, new Timestamp(Instant.now.toEpochMilli + expiry.toMillis.toLong))
+    authTokenDAO.save(token).flatMap {
+      case _ => Future.successful(tokenID)
+    }
+  }
+
+  /**
+   * Validates a token ID.
+   *
+   * @param id The token ID to validate.
+   * @return The token if it's valid, None otherwise.
+   */
+  def validate(id: UUID) = authTokenDAO.find(id).flatMap {
```

Comment:
If the function is longer than one line, make sure to have it in this format:
```
  def validate(id: UUID) = {
    ...
  }
```
[Reference](https://docs.scala-lang.org/style/declarations.html)

Reply:
Got it